File: CS/Form1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Helpers;


namespace RefreshHelperApp
{
    /// <summary>
    /// Summary description for Form1.
    /// </summary>
    public class Form1 : System.Windows.Forms.Form
    {
        private DevExpress.XtraGrid.GridControl gridControl1;
        private DevExpress.XtraGrid.Views.Grid.GridView gridView1;
        private DevExpress.XtraEditors.SimpleButton simpleButton1;
        private DevExpress.XtraEditors.SimpleButton simpleButton2;
        private DevExpress.XtraGrid.Views.Grid.GridView gridView2;
        private DevExpress.XtraGrid.Columns.GridColumn colCustomerID1;
        private DevExpress.XtraGrid.Columns.GridColumn colEmployeeID;
        private DevExpress.XtraGrid.Columns.GridColumn colFreight;
        private DevExpress.XtraGrid.Columns.GridColumn colOrderDate;
        private DevExpress.XtraGrid.Columns.GridColumn colOrderID;
        private DevExpress.XtraGrid.Columns.GridColumn colRequiredDate;
        private DevExpress.XtraGrid.Columns.GridColumn colShipAddress;
        private DevExpress.XtraGrid.Columns.GridColumn colShipCity;
        private DevExpress.XtraGrid.Columns.GridColumn colShipCountry;
        private DevExpress.XtraGrid.Columns.GridColumn colShipName;
        private DevExpress.XtraGrid.Columns.GridColumn colShippedDate;
        private DevExpress.XtraGrid.Columns.GridColumn colShipPostalCode;
        private DevExpress.XtraGrid.Columns.GridColumn colShipRegion;
        private DevExpress.XtraGrid.Columns.GridColumn colShipVia;
        private System.Windows.Forms.Panel panel1;
        private IContainer components;

        public Form1()
        {
            //
            // Required for Windows Form Designer support
            //
            InitializeComponent();

            //
            // TODO: Add any constructor code after InitializeComponent call
            //
        }

        /// <summary>
        /// Clean up any resources being used.
        /// </summary>
        protected override void Dispose(bool disposing)
        {
            if (disposing)
            {
                if (components != null)
                {
                    components.Dispose();
                }
            }
            base.Dispose(disposing);
        }

        #region Windows Form Designer generated code
        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
            System.ComponentModel.ComponentResourceManager resources = new System.ComponentModel.ComponentResourceManager(typeof(Form1));
            DevExpress.XtraGrid.GridLevelNode gridLevelNode1 = new DevExpress.XtraGrid.GridLevelNode();
            this.gridView2 = new DevExpress.XtraGrid.Views.Grid.GridView();
            this.colCustomerID1 = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colEmployeeID = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colFreight = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colOrderDate = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colOrderID = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colRequiredDate = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colShipAddress = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colShipCity = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colShipCountry = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colShipName = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colShippedDate = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colShipPostalCode = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colShipRegion = new DevExpress.XtraGrid.Columns.GridColumn();
            this.colShipVia = new DevExpress.XtraGrid.Columns.GridColumn();
            this.gridControl1 = new DevExpress.XtraGrid.GridControl();
            this.gridView1 = new DevExpress.XtraGrid.Views.Grid.GridView();
            this.simpleButton1 = new DevExpress.XtraEditors.SimpleButton();
            this.simpleButton2 = new DevExpress.XtraEditors.SimpleButton();
            this.panel1 = new System.Windows.Forms.Panel();
            ((System.ComponentModel.ISupportInitialize)(this.gridView2)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.gridControl1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.gridView1)).BeginInit();
            this.panel1.SuspendLayout();
            this.SuspendLayout();
            // 
            // gridView2
            // 
            this.gridView2.Columns.AddRange(new DevExpress.XtraGrid.Columns.GridColumn[] {
            this.colCustomerID1,
            this.colEmployeeID,
            this.colFreight,
            this.colOrderDate,
            this.colOrderID,
            this.colRequiredDate,
            this.colShipAddress,
            this.colShipCity,
            this.colShipCountry,
            this.colShipName,
            this.colShippedDate,
            this.colShipPostalCode,
            this.colShipRegion,
            this.colShipVia});
            this.gridView2.GridControl = this.gridControl1;
            this.gridView2.Name = "gridView2";
            // 
            // colCustomerID1
            // 
            this.colCustomerID1.Caption = "CustomerID";
            this.colCustomerID1.FieldName = "CustomerID";
            this.colCustomerID1.Name = "colCustomerID1";
            this.colCustomerID1.Visible = true;
            this.colCustomerID1.VisibleIndex = 0;
            // 
            // colEmployeeID
            // 
            this.colEmployeeID.Caption = "EmployeeID";
            this.colEmployeeID.FieldName = "EmployeeID";
            this.colEmployeeID.Name = "colEmployeeID";
            this.colEmployeeID.Visible = true;
            this.colEmployeeID.VisibleIndex = 1;
            // 
            // colFreight
            // 
            this.colFreight.Caption = "Freight";
            this.colFreight.FieldName = "Freight";
            this.colFreight.Name = "colFreight";
            this.colFreight.Visible = true;
            this.colFreight.VisibleIndex = 2;
            // 
            // colOrderDate
            // 
            this.colOrderDate.Caption = "OrderDate";
            this.colOrderDate.FieldName = "OrderDate";
            this.colOrderDate.Name = "colOrderDate";
            this.colOrderDate.Visible = true;
            this.colOrderDate.VisibleIndex = 3;
            // 
            // colOrderID
            // 
            this.colOrderID.Caption = "OrderID";
            this.colOrderID.FieldName = "OrderID";
            this.colOrderID.Name = "colOrderID";
            this.colOrderID.Visible = true;
            this.colOrderID.VisibleIndex = 4;
            // 
            // colRequiredDate
            // 
            this.colRequiredDate.Caption = "RequiredDate";
            this.colRequiredDate.FieldName = "RequiredDate";
            this.colRequiredDate.Name = "colRequiredDate";
            this.colRequiredDate.Visible = true;
            this.colRequiredDate.VisibleIndex = 5;
            // 
            // colShipAddress
            // 
            this.colShipAddress.Caption = "ShipAddress";
            this.colShipAddress.FieldName = "ShipAddress";
            this.colShipAddress.Name = "colShipAddress";
            this.colShipAddress.Visible = true;
            this.colShipAddress.VisibleIndex = 6;
            // 
            // colShipCity
            // 
            this.colShipCity.Caption = "ShipCity";
            this.colShipCity.FieldName = "ShipCity";
            this.colShipCity.Name = "colShipCity";
            this.colShipCity.Visible = true;
            this.colShipCity.VisibleIndex = 7;
            // 
            // colShipCountry
            // 
            this.colShipCountry.Caption = "ShipCountry";
            this.colShipCountry.FieldName = "ShipCountry";
            this.colShipCountry.Name = "colShipCountry";
            this.colShipCountry.Visible = true;
            this.colShipCountry.VisibleIndex = 8;
            // 
            // colShipName
            // 
            this.colShipName.Caption = "ShipName";
            this.colShipName.FieldName = "ShipName";
            this.colShipName.Name = "colShipName";
            this.colShipName.Visible = true;
            this.colShipName.VisibleIndex = 9;
            // 
            // colShippedDate
            // 
            this.colShippedDate.Caption = "ShippedDate";
            this.colShippedDate.FieldName = "ShippedDate";
            this.colShippedDate.Name = "colShippedDate";
            this.colShippedDate.Visible = true;
            this.colShippedDate.VisibleIndex = 10;
            // 
            // colShipPostalCode
            // 
            this.colShipPostalCode.Caption = "ShipPostalCode";
            this.colShipPostalCode.FieldName = "ShipPostalCode";
            this.colShipPostalCode.Name = "colShipPostalCode";
            this.colShipPostalCode.Visible = true;
            this.colShipPostalCode.VisibleIndex = 11;
            // 
            // colShipRegion
            // 
            this.colShipRegion.Caption = "ShipRegion";
            this.colShipRegion.FieldName = "ShipRegion";
            this.colShipRegion.Name = "colShipRegion";
            this.colShipRegion.Visible = true;
            this.colShipRegion.VisibleIndex = 12;
            // 
            // colShipVia
            // 
            this.colShipVia.Caption = "ShipVia";
            this.colShipVia.FieldName = "ShipVia";
            this.colShipVia.Name = "colShipVia";
            this.colShipVia.Visible = true;
            this.colShipVia.VisibleIndex = 13;
            // 
            // gridControl1
            // 
            this.gridControl1.Dock = System.Windows.Forms.DockStyle.Fill;
            gridLevelNode1.LevelTemplate = this.gridView2;
            gridLevelNode1.RelationName = "CustomersOrders";
            this.gridControl1.LevelTree.Nodes.AddRange(new DevExpress.XtraGrid.GridLevelNode[] {
            gridLevelNode1});
            this.gridControl1.Location = new System.Drawing.Point(0, 40);
            this.gridControl1.MainView = this.gridView1;
            this.gridControl1.Name = "gridControl1";
            this.gridControl1.Size = new System.Drawing.Size(875, 510);
            this.gridControl1.TabIndex = 0;
            this.gridControl1.ViewCollection.AddRange(new DevExpress.XtraGrid.Views.Base.BaseView[] {
            this.gridView1,
            this.gridView2});
            // 
            // gridView1
            // 
            this.gridView1.GridControl = this.gridControl1;
            this.gridView1.Name = "gridView1";
            this.gridView1.OptionsSelection.MultiSelect = true;
            // 
            // simpleButton1
            // 
            this.simpleButton1.Location = new System.Drawing.Point(8, 8);
            this.simpleButton1.Name = "simpleButton1";
            this.simpleButton1.Size = new System.Drawing.Size(75, 23);
            this.simpleButton1.TabIndex = 1;
            this.simpleButton1.Text = "Save";
            this.simpleButton1.Click += new System.EventHandler(this.simpleButton1_Click);
            // 
            // simpleButton2
            // 
            this.simpleButton2.Location = new System.Drawing.Point(104, 8);
            this.simpleButton2.Name = "simpleButton2";
            this.simpleButton2.Size = new System.Drawing.Size(72, 24);
            this.simpleButton2.TabIndex = 2;
            this.simpleButton2.Text = "Load";
            this.simpleButton2.Click += new System.EventHandler(this.simpleButton2_Click);
            // 
            // panel1
            // 
            this.panel1.Controls.Add(this.simpleButton1);
            this.panel1.Controls.Add(this.simpleButton2);
            this.panel1.Dock = System.Windows.Forms.DockStyle.Top;
            this.panel1.Location = new System.Drawing.Point(0, 0);
            this.panel1.Name = "panel1";
            this.panel1.Size = new System.Drawing.Size(875, 40);
            this.panel1.TabIndex = 3;
            // 
            // Form1
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.ClientSize = new System.Drawing.Size(875, 550);
            this.Controls.Add(this.gridControl1);
            this.Controls.Add(this.panel1);
            this.Name = "Form1";
            this.Text = "Form1";
            this.Load += new System.EventHandler(this.Form1_Load);
            ((System.ComponentModel.ISupportInitialize)(this.gridView2)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.gridControl1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.gridView1)).EndInit();
            this.panel1.ResumeLayout(false);
            this.ResumeLayout(false);

        }
        #endregion

        /// <summary>
        /// The main entry point for the application.
        /// </summary>
        [STAThread]
        static void Main()
        {
            Application.Run(new Form1());
        }

        RefreshHelper helper;

        private void Form1_Load(object sender, System.EventArgs e)
        {
            helper = new RefreshHelper(gridView1, "ID");

            DataSet dataSet11 = new DataSet();
            dataSet11.Tables.Add(GetCustomerDataTable());
            dataSet11.Tables.Add(GetPersonDataTable());
            DataColumn keyColumn = dataSet11.Tables["Customers"].Columns["ID"];
            DataColumn foreignKeyColumn = dataSet11.Tables["Persons"].Columns["ID"];
            dataSet11.Relations.Add("CustomersPersons", keyColumn, foreignKeyColumn);
            gridControl1.DataSource = dataSet11.Tables["Customers"];
            gridControl1.ForceInitialize();
        }

        private void simpleButton1_Click(object sender, System.EventArgs e)
        {
            helper.SaveViewInfo();
        }

        private void simpleButton2_Click(object sender, System.EventArgs e)
        {
            helper.LoadViewInfo();
        }
        DataTable GetCustomerDataTable()
        {
            DataTable table = new DataTable();
            table.TableName = "Customers";
            table.Columns.Add(new DataColumn("Items", typeof(string)));
            table.Columns.Add(new DataColumn("Money", typeof(double)));
            table.Columns.Add(new DataColumn("ID", typeof(int)));
            for (int i = 0; i < 10; i++)
                table.Rows.Add("Product " + i, 3000 + i * 298.55M, i);
            return table;
        }
        DataTable GetPersonDataTable()
        {
            DataTable table = new DataTable();
            table.TableName = "Persons";
            table.Columns.Add(new DataColumn("FirstName", typeof(string)));
            table.Columns.Add(new DataColumn("SecondName", typeof(string)));
            table.Columns.Add(new DataColumn("Age", typeof(int)));
            table.Columns.Add(new DataColumn("ID", typeof(int)));
            for (int i = 0; i < 50; i++)
            {
                string name = "Adam";
                string secondName = "Smith";
                if (i % 2 == 0)
                {
                    name = "Ben";
                    secondName = "Black";
                }
                table.Rows.Add(name, secondName, 20 + i / 2, i % 10);
            }
            return table;
        }
    }
}

File: VB/Form1.vb
Imports System
Imports System.Drawing
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Data
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Helpers

Namespace RefreshHelperApp

    ''' <summary>
    ''' Summary description for Form1.
    ''' </summary>
    Public Class Form1
        Inherits Form

        Private gridControl1 As DevExpress.XtraGrid.GridControl

        Private gridView1 As GridView

        Private simpleButton1 As DevExpress.XtraEditors.SimpleButton

        Private simpleButton2 As DevExpress.XtraEditors.SimpleButton

        Private gridView2 As GridView

        Private colCustomerID1 As DevExpress.XtraGrid.Columns.GridColumn

        Private colEmployeeID As DevExpress.XtraGrid.Columns.GridColumn

        Private colFreight As DevExpress.XtraGrid.Columns.GridColumn

        Private colOrderDate As DevExpress.XtraGrid.Columns.GridColumn

        Private colOrderID As DevExpress.XtraGrid.Columns.GridColumn

        Private colRequiredDate As DevExpress.XtraGrid.Columns.GridColumn

        Private colShipAddress As DevExpress.XtraGrid.Columns.GridColumn

        Private colShipCity As DevExpress.XtraGrid.Columns.GridColumn

        Private colShipCountry As DevExpress.XtraGrid.Columns.GridColumn

        Private colShipName As DevExpress.XtraGrid.Columns.GridColumn

        Private colShippedDate As DevExpress.XtraGrid.Columns.GridColumn

        Private colShipPostalCode As DevExpress.XtraGrid.Columns.GridColumn

        Private colShipRegion As DevExpress.XtraGrid.Columns.GridColumn

        Private colShipVia As DevExpress.XtraGrid.Columns.GridColumn

        Private panel1 As Panel

        Private components As IContainer

        Public Sub New()
            '
            ' Required for Windows Form Designer support
            '
            InitializeComponent()
        '
        ' TODO: Add any constructor code after InitializeComponent call
        '
        End Sub

        ''' <summary>
        ''' Clean up any resources being used.
        ''' </summary>
        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If components IsNot Nothing Then
                    components.Dispose()
                End If
            End If

            MyBase.Dispose(disposing)
        End Sub

#Region "Windows Form Designer generated code"
        ''' <summary>
        ''' Required method for Designer support - do not modify
        ''' the contents of this method with the code editor.
        ''' </summary>
        Private Sub InitializeComponent()
            Dim resources As System.ComponentModel.ComponentResourceManager = New System.ComponentModel.ComponentResourceManager(GetType(Form1))
            Dim gridLevelNode1 As DevExpress.XtraGrid.GridLevelNode = New DevExpress.XtraGrid.GridLevelNode()
            gridView2 = New GridView()
            colCustomerID1 = New DevExpress.XtraGrid.Columns.GridColumn()
            colEmployeeID = New DevExpress.XtraGrid.Columns.GridColumn()
            colFreight = New DevExpress.XtraGrid.Columns.GridColumn()
            colOrderDate = New DevExpress.XtraGrid.Columns.GridColumn()
            colOrderID = New DevExpress.XtraGrid.Columns.GridColumn()
            colRequiredDate = New DevExpress.XtraGrid.Columns.GridColumn()
            colShipAddress = New DevExpress.XtraGrid.Columns.GridColumn()
            colShipCity = New DevExpress.XtraGrid.Columns.GridColumn()
            colShipCountry = New DevExpress.XtraGrid.Columns.GridColumn()
            colShipName = New DevExpress.XtraGrid.Columns.GridColumn()
            colShippedDate = New DevExpress.XtraGrid.Columns.GridColumn()
            colShipPostalCode = New DevExpress.XtraGrid.Columns.GridColumn()
            colShipRegion = New DevExpress.XtraGrid.Columns.GridColumn()
            colShipVia = New DevExpress.XtraGrid.Columns.GridColumn()
            gridControl1 = New DevExpress.XtraGrid.GridControl()
            gridView1 = New GridView()
            simpleButton1 = New DevExpress.XtraEditors.SimpleButton()
            simpleButton2 = New DevExpress.XtraEditors.SimpleButton()
            panel1 = New Panel()
            CType(gridView2, System.ComponentModel.ISupportInitialize).BeginInit()
            CType(gridControl1, System.ComponentModel.ISupportInitialize).BeginInit()
            CType(gridView1, System.ComponentModel.ISupportInitialize).BeginInit()
            panel1.SuspendLayout()
            Me.SuspendLayout()
            ' 
            ' gridView2
            ' 
            gridView2.Columns.AddRange(New DevExpress.XtraGrid.Columns.GridColumn() {colCustomerID1, colEmployeeID, colFreight, colOrderDate, colOrderID, colRequiredDate, colShipAddress, colShipCity, colShipCountry, colShipName, colShippedDate, colShipPostalCode, colShipRegion, colShipVia})
            gridView2.GridControl = gridControl1
            gridView2.Name = "gridView2"
            ' 
            ' colCustomerID1
            ' 
            colCustomerID1.Caption = "CustomerID"
            colCustomerID1.FieldName = "CustomerID"
            colCustomerID1.Name = "colCustomerID1"
            colCustomerID1.Visible = True
            colCustomerID1.VisibleIndex = 0
            ' 
            ' colEmployeeID
            ' 
            colEmployeeID.Caption = "EmployeeID"
            colEmployeeID.FieldName = "EmployeeID"
            colEmployeeID.Name = "colEmployeeID"
            colEmployeeID.Visible = True
            colEmployeeID.VisibleIndex = 1
            ' 
            ' colFreight
            ' 
            colFreight.Caption = "Freight"
            colFreight.FieldName = "Freight"
            colFreight.Name = "colFreight"
            colFreight.Visible = True
            colFreight.VisibleIndex = 2
            ' 
            ' colOrderDate
            ' 
            colOrderDate.Caption = "OrderDate"
            colOrderDate.FieldName = "OrderDate"
            colOrderDate.Name = "colOrderDate"
            colOrderDate.Visible = True
            colOrderDate.VisibleIndex = 3
            ' 
            ' colOrderID
            ' 
            colOrderID.Caption = "OrderID"
            colOrderID.FieldName = "OrderID"
            colOrderID.Name = "colOrderID"
            colOrderID.Visible = True
            colOrderID.VisibleIndex = 4
            ' 
            ' colRequiredDate
            ' 
            colRequiredDate.Caption = "RequiredDate"
            colRequiredDate.FieldName = "RequiredDate"
            colRequiredDate.Name = "colRequiredDate"
            colRequiredDate.Visible = True
            colRequiredDate.VisibleIndex = 5
            ' 
            ' colShipAddress
            ' 
            colShipAddress.Caption = "ShipAddress"
            colShipAddress.FieldName = "ShipAddress"
            colShipAddress.Name = "colShipAddress"
            colShipAddress.Visible = True
            colShipAddress.VisibleIndex = 6
            ' 
            ' colShipCity
            ' 
            colShipCity.Caption = "ShipCity"
            colShipCity.FieldName = "ShipCity"
            colShipCity.Name = "colShipCity"
            colShipCity.Visible = True
            colShipCity.VisibleIndex = 7
            ' 
            ' colShipCountry
            ' 
            colShipCountry.Caption = "ShipCountry"
            colShipCountry.FieldName = "ShipCountry"
            colShipCountry.Name = "colShipCountry"
            colShipCountry.Visible = True
            colShipCountry.VisibleIndex = 8
            ' 
            ' colShipName
            ' 
            colShipName.Caption = "ShipName"
            colShipName.FieldName = "ShipName"
            colShipName.Name = "colShipName"
            colShipName.Visible = True
            colShipName.VisibleIndex = 9
            ' 
            ' colShippedDate
            ' 
            colShippedDate.Caption = "ShippedDate"
            colShippedDate.FieldName = "ShippedDate"
            colShippedDate.Name = "colShippedDate"
            colShippedDate.Visible = True
            colShippedDate.VisibleIndex = 10
            ' 
            ' colShipPostalCode
            ' 
            colShipPostalCode.Caption = "ShipPostalCode"
            colShipPostalCode.FieldName = "ShipPostalCode"
            colShipPostalCode.Name = "colShipPostalCode"
            colShipPostalCode.Visible = True
            colShipPostalCode.VisibleIndex = 11
            ' 
            ' colShipRegion
            ' 
            colShipRegion.Caption = "ShipRegion"
            colShipRegion.FieldName = "ShipRegion"
            colShipRegion.Name = "colShipRegion"
            colShipRegion.Visible = True
            colShipRegion.VisibleIndex = 12
            ' 
            ' colShipVia
            ' 
            colShipVia.Caption = "ShipVia"
            colShipVia.FieldName = "ShipVia"
            colShipVia.Name = "colShipVia"
            colShipVia.Visible = True
            colShipVia.VisibleIndex = 13
            ' 
            ' gridControl1
            ' 
            gridControl1.Dock = DockStyle.Fill
            gridLevelNode1.LevelTemplate = gridView2
            gridLevelNode1.RelationName = "CustomersOrders"
            gridControl1.LevelTree.Nodes.AddRange(New DevExpress.XtraGrid.GridLevelNode() {gridLevelNode1})
            gridControl1.Location = New System.Drawing.Point(0, 40)
            gridControl1.MainView = gridView1
            gridControl1.Name = "gridControl1"
            gridControl1.Size = New System.Drawing.Size(875, 510)
            gridControl1.TabIndex = 0
            gridControl1.ViewCollection.AddRange(New DevExpress.XtraGrid.Views.Base.BaseView() {gridView1, gridView2})
            ' 
            ' gridView1
            ' 
            gridView1.GridControl = gridControl1
            gridView1.Name = "gridView1"
            gridView1.OptionsSelection.MultiSelect = True
            ' 
            ' simpleButton1
            ' 
            simpleButton1.Location = New System.Drawing.Point(8, 8)
            simpleButton1.Name = "simpleButton1"
            simpleButton1.Size = New System.Drawing.Size(75, 23)
            simpleButton1.TabIndex = 1
            simpleButton1.Text = "Save"
            AddHandler simpleButton1.Click, New EventHandler(AddressOf simpleButton1_Click)
            ' 
            ' simpleButton2
            ' 
            simpleButton2.Location = New System.Drawing.Point(104, 8)
            simpleButton2.Name = "simpleButton2"
            simpleButton2.Size = New System.Drawing.Size(72, 24)
            simpleButton2.TabIndex = 2
            simpleButton2.Text = "Load"
            AddHandler simpleButton2.Click, New EventHandler(AddressOf simpleButton2_Click)
            ' 
            ' panel1
            ' 
            panel1.Controls.Add(simpleButton1)
            panel1.Controls.Add(simpleButton2)
            panel1.Dock = DockStyle.Top
            panel1.Location = New System.Drawing.Point(0, 0)
            panel1.Name = "panel1"
            panel1.Size = New System.Drawing.Size(875, 40)
            panel1.TabIndex = 3
            ' 
            ' Form1
            ' 
            AutoScaleBaseSize = New System.Drawing.Size(5, 13)
            ClientSize = New System.Drawing.Size(875, 550)
            Me.Controls.Add(gridControl1)
            Me.Controls.Add(panel1)
            Name = "Form1"
            Text = "Form1"
            AddHandler Load, New EventHandler(AddressOf Form1_Load)
            CType(gridView2, System.ComponentModel.ISupportInitialize).EndInit()
            CType(gridControl1, System.ComponentModel.ISupportInitialize).EndInit()
            CType(gridView1, System.ComponentModel.ISupportInitialize).EndInit()
            panel1.ResumeLayout(False)
            Me.ResumeLayout(False)
        End Sub

#End Region
        ''' <summary>
        ''' The main entry point for the application.
        ''' </summary>
        <STAThread>
        Shared Sub Main()
            Call Application.Run(New Form1())
        End Sub

        Private helper As RefreshHelper

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            helper = New RefreshHelper(gridView1, "ID")
            Dim dataSet11 As DataSet = New DataSet()
            dataSet11.Tables.Add(GetCustomerDataTable())
            dataSet11.Tables.Add(GetPersonDataTable())
            Dim keyColumn As DataColumn = dataSet11.Tables("Customers").Columns("ID")
            Dim foreignKeyColumn As DataColumn = dataSet11.Tables("Persons").Columns("ID")
            dataSet11.Relations.Add("CustomersPersons", keyColumn, foreignKeyColumn)
            gridControl1.DataSource = dataSet11.Tables("Customers")
            gridControl1.ForceInitialize()
        End Sub

        Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs)
            helper.SaveViewInfo()
        End Sub

        Private Sub simpleButton2_Click(ByVal sender As Object, ByVal e As EventArgs)
            helper.LoadViewInfo()
        End Sub

        Private Function GetCustomerDataTable() As DataTable
            Dim table As DataTable = New DataTable()
            table.TableName = "Customers"
            table.Columns.Add(New DataColumn("Items", GetType(String)))
            table.Columns.Add(New DataColumn("Money", GetType(Double)))
            table.Columns.Add(New DataColumn("ID", GetType(Integer)))
            For i As Integer = 0 To 10 - 1
                table.Rows.Add("Product " & i, 3000 + i * 298.55D, i)
            Next

            Return table
        End Function

        Private Function GetPersonDataTable() As DataTable
            Dim table As DataTable = New DataTable()
            table.TableName = "Persons"
            table.Columns.Add(New DataColumn("FirstName", GetType(String)))
            table.Columns.Add(New DataColumn("SecondName", GetType(String)))
            table.Columns.Add(New DataColumn("Age", GetType(Integer)))
            table.Columns.Add(New DataColumn("ID", GetType(Integer)))
            For i As Integer = 0 To 50 - 1
                Dim name As String = "Adam"
                Dim secondName As String = "Smith"
                If i Mod 2 = 0 Then
                    name = "Ben"
                    secondName = "Black"
                End If

                table.Rows.Add(name, secondName, 20 + i \ 2, i Mod 10)
            Next

            Return table
        End Function
    End Class
End Namespace
